fix(designer): Fix invoker connection toggle behavior

bnarayanared opened the pull request:
Before:
By default the invoker connection toggle was on:
![image](https://user-images.githubusercontent.com/91710207/235505581-23addff2-622e-4f6f-acdf-ab4e20844de7.png)


After the fix:
By default the "Use Invoker's Connection" toggle is off:
![image](https://user-images.githubusercontent.com/91710207/235505611-0101cf84-9447-4036-a43a-3ecf812aa1e7.png)

preetriti1 commented on libs/designer/src/lib/core/actions/bjsworkflow/settings.ts:
```diff
@@ -877,16 +873,16 @@ const isInvokerConnectionSupported = (isTrigger: boolean, nodeType: string, root
-    dispatch?.(addInvokerSupport({ connectionReferences }));
+  if (
+    isInvokerConnectionSupported(isTrigger, nodeType, rootNodeId) &&
+    connectorId === Constants.INVOKER_CONNECTION.DATAVERSE_CONNECTOR_ID
```
Please do not add arm connectorid checks in the code. Add metadata in the manifest and manifest metadatas can be looked at but not connectorIds, we do not want such checks in the client code.
You can patch this connectors manifest in your client service to provide appropriate settings, if this is manifest based connector please patch it in your client service to add proper setting metadata for invokerconnection support.

preetriti1 replied:
If there was a previous PR which checked in such constants, it will be removed because we do not want to follow this kind of model which was done in old designer.

bnarayanared replied:
Removed the check for `connectorId`

preetriti1 commented on libs/designer/src/lib/core/actions/bjsworkflow/operationdeserializer.ts:
```diff
@@ -190,9 +189,7 @@ export const initializeOperationDetailsForManifest = async (
-        rootNodeId,
-        connectionReferences,
-        dispatch
+        rootNodeId
```
Does this not require any changes in your settings file?

bnarayanared replied:
It is already added as an optional property

preetriti1 commented:
You are missing changes in core/actions/bjsworkflow/setting.ts to remove the connectorId check from isInvokerConnectionSupported method.

bnarayanared commented:
> You are missing changes in core/actions/bjsworkflow/setting.ts to remove the connectorId check from isInvokerConnectionSupported method.

Removed and updated branch with required settings

preetriti1 commented on libs/designer/src/lib/core/actions/bjsworkflow/add.ts:
```diff
@@ -332,11 +329,16 @@ const getOperationType = (operation: DiscoveryOperation<DiscoveryResultTypes>):
 
-export const getRootNodeDetails = (workflowState: WorkflowState, connectorId: string): string | undefined => {
-  if (workflowState.graph?.children && connectorId.indexOf(Constants.INVOKER_CONNECTION.DATAVERSE_CONNECTOR_ID) > -1) {
+export const getRootNodeManifest = async (
```
rename to getTriggerNodeManifest, please do not use the term root here unless you want roots in child graphs

preetriti1 commented on libs/designer/src/lib/core/actions/bjsworkflow/add.ts:
```diff
@@ -332,11 +329,16 @@ const getOperationType = (operation: DiscoveryOperation<DiscoveryResultTypes>):
+  operations: OperationMetadataState
+): Promise<OperationManifest | undefined> => {
+  if (workflowState?.graph?.children !== undefined) {
     for (const child of workflowState.graph.children) {
       if (workflowState.graph?.id === 'root') {
-        return child.id;
+        const rootNodeOperationInfo = operations.operationInfo[child.id];
+        const { connectorId, operationId } = rootNodeOperationInfo;
+        return await getOperationManifest({ connectorId, operationId });
       }
     }
```
Update code to
`const triggerNodeId = getTriggerNodeId(workflowState);
 const operationInfo = operations.operationInfo[triggerNodeId]
 if (operationInfo && OperationManifestService().isSupported(operationInfo.type, operationInfo.kind)) {
     return getOperationManifest(operationInfo.connectorId, operationInfo.operationId);
 }
 return undefined;
 `

preetriti1 commented on libs/designer/src/lib/core/actions/bjsworkflow/add.ts:
```diff
@@ -205,25 +210,17 @@ const initializeOperationDetails = async (
-  );
+  const settings = getOperationSettings(isTrigger, operationInfo, initData.nodeOutputs, manifest, swagger, operation);
   dispatch(updateNodeSettings({ id: nodeId, settings }));
+  updateTriggerNodeManifestForInvokerSettings(isTrigger, rootNodeManifest, nodeId, dispatch);
```
if (triggerNodeManifest) { updateInvokerSettings()}

bnarayanared commented on libs/designer/src/lib/core/actions/bjsworkflow/initialize.ts:
```diff
@@ -442,3 +443,19 @@ export const getCustomSwaggerIfNeeded = async (
+  dispatch: Dispatch
+): void => {
+  if (isTrigger) {
+    const invokerSetting = { isSupported: false, value: { enabled: false } };
```
Added default values to settings

preetriti1 commented:
You are missing changes to remove the connectorId constant - 
INVOKER_CONNECTION: {
    DATAVERSE_CUD_TRIGGER: 'When_a_row_is_added,_modified_or_deleted',
    DATAVERSE_CONNECTOR_ID: '/providers/Microsoft.PowerApps/apis/shared_commondataserviceforapps',
  },
  
  Also, we should never reupdate the connectionReferences sent to designer from host. Please remove the method - addInvokerSupport from connectionslice.
  No where in the code you will find any host doing it, host should be modifying the connection reference if needed before sending it to the designer.
  manage connection references in the portal code before sending it to designer, in logic apps portal we also do that.

preetriti1 reviewed (changes requested):
connectorId removal is still pending

bnarayanared commented:
> connectorId removal is still pending

Removed
